src/containers/EnterpriseApp/index.jsx

```diff
@@ -9,17 +9,38 @@ import AdminPage from '../AdminPage';
 import NotFoundPage from '../NotFoundPage';
 import ErrorPage from '../ErrorPage';
 
-import { fetchPortalConfiguration } from '../../data/actions/portalConfiguration';
+import fetchEnterpriseList from '../../data/actions/enterpriseList';
+import { setPortalConfiguration } from '../../data/actions/portalConfiguration';
 import { getLocalUser } from '../../data/actions/authentication';
 
 class EnterpriseApp extends React.Component {
   componentDidMount() {
     const { enterpriseSlug } = this.props.match.params;
+    if (!this.props.enterprises) {
+      this.props.fetchEnterpriseList();
```

Comment:
With this refactor, we are getting the enterprise branding configuration from the API response returned in the `with_access_to` endpoint. On a hard refresh of the dashboard page, we never call that endpoint so we need to do that here if no enterprises are loaded yet.